### ISSUE TITLE
Dispatcher & Response don't handle mixed-type content very well

### ISSUE THREAD

**Issue by robin-mollie** [devvoh/parable#41]
`Dispatcher::97`:

`$content = $this->response->returnOutputBuffer() . $content;`

This does not handle array data. Make it:

```
$bufferContent = $this->response->returnAllOutputBuffers();
if (!empty($bufferContent)) {
    $this->response->appendContent($bufferContent);
}
```

This still appends it _before_ the provided `$content`, but lets `Response` handle the data type.

HOWEVER. `Response::248`:

```
public function appendContent($content)
{
    if (!empty($content)) {
        if (is_array($this->content)) {
            $this->content[] = $content;
        } else {
            $this->content .= $content;
        }
    }
    return $this;
}
```

So okay, cool, we check the type of `$this->content` and append appropriately.

But what if `$this->content` is a string but `$content` is an array?

Quick fix: take `$this->content`'s string value, do `$this->setContent([$this->content]);` and continue along. From that point on, all content is seen as array values.

Even quicker and more consistent fix: Always store content in an array and just add new key/value pairs always. If `setContent()` is called with a string, pack it into a new array and _boom_, consistency.

Mixed types bad.

**Comment by devvoh** [comment]
Great suggestion, other me!

**Comment by devvoh** [comment]
Terrible suggestions, other me. The consequences of approaching the problem as mentioned are too grand, sadly. They involve a different way of thinking about the content and especially when working with Json output, does not work as it should.

The fix will be the following. Only change to Response is that it'll get `returnOutput()` to return the output class currently set.

In `Dispatcher`, I will check whether `$content` is an array and if the output class is Json. If so, Parable will set it on the response directly, ignoring output buffers.

If not the above and `$content` is not a string, we throw an exception: `"Route returned value of type '{$type}', which output class '{$outputClassName}' cannot handle."`

Any other situation, we act as usual and concatenate all strings together.

**Comment by devvoh** [comment]
Fixed in 5b0ada4fe1159af5903104cf69668f0556785329, will be released in 1.1.0.
